pkg/kubelet/deviceplugin/manager.go

```
@@ -74,15 +74,25 @@ func removeContents(dir string) error {
 		return err
 	}
 	for _, name := range names {
-		// TODO: skip checkpoint file and check for file type.
-		err = os.RemoveAll(filepath.Join(dir, name))
+		filePath := filepath.Join(dir, name)
+		if filePath == m.CheckpointFile() {
+			continue
+		}
+		if stat, err := os.Stat(filePath); err == nil && stat.IsDir() {
```

stat err is silently ignored here.

---

good catch. fixed.